lms/djangoapps/grades/models.py

```diff
@@ -0,0 +1,250 @@
+"""
+Models used for robust grading.
+
+Robust grading allows student scores to be saved per-subsection independent
+of any changes that may occur to the course after the score is achieved.
+"""
+
+from base64 import b64encode
+from collections import namedtuple
+from hashlib import sha256
+import json
+import logging
+from operator import attrgetter
+
+from django.db import models
+from django.db.utils import IntegrityError
+from model_utils.models import TimeStampedModel
+
+from coursewarehistoryextended.fields import UnsignedBigIntAutoField
+from xmodule_django.models import CourseKeyField, UsageKeyField
+
+
+log = logging.getLogger(__name__)
+
+
+# Used to serialize information about a block at the time it was used in
+# grade calculation.
+BlockRecord = namedtuple('BlockRecord', ['locator', 'weight', 'max_score'])
+
+
+def blockrecord_list_to_json(blocks):
+    """
+    Return a JSON-serialized version of the list of block records, using a
+    stable ordering.
+    """
+    sorted_blocks = sorted([block._asdict() for block in blocks], key=attrgetter('locator'))
+    return json.dumps(sorted_blocks, separators=(',', ':'), sort_keys=True)
```

Comment:
Yes.  sort_keys refers to the keys in each dictionary.  sorted_blocks refers to the order of the dictionaries within the BlockRecordList.